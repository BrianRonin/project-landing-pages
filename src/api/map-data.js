import { mapMenu } from './menu/map-menu'
import { mapSections } from './sections/map-sections'

export const mapData = (api = {}) => {
  const { data: pagesData = [{}] } = api

  return pagesData.map((data) => {
    const { attributes = {} } = data
    const {
      footer_text: footerHtml = '',
      slug = '',
      title = '',
      sections = [],
      menu = {},
    } = attributes

    const x = {
      footerHtml,
      slug,
      title,
      sections: mapSections(sections),
      menu: mapMenu(menu),
    }

    return x
  })
}
