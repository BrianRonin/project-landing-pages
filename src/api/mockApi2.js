export const mockApi = {
  data: [
    {
      id: 2,
      attributes: {
        title: 'minha pagina',
        slug: 'minha-pagina',
        createdAt: '2022-08-26T05:06:02.113Z',
        updatedAt: '2022-08-26T20:23:28.556Z',
        publishedAt: '2022-08-26T05:10:46.365Z',
        footer_text: 'feito com carinho por BrianRonin',
        sections: [
          {
            id: 1,
            __component: 'section.section-two-columns',
            title: 'titulo qualquer',
            description: 'uma breve desc',
            image: {
              data: {
                id: 2,
                attributes: {
                  name: 'analytics_re_wxt8.svg',
                  alternativeText: 'analytics_re_wxt8.svg',
                  caption: 'analytics_re_wxt8.svg',
                  width: 363,
                  height: 438,
                  formats: null,
                  hash: 'analytics_re_wxt8_1920e22fe3',
                  ext: '.svg',
                  mime: 'image/svg+xml',
                  size: 7.3,
                  url: 'https://res.cloudinary.com/brianronin/image/upload/v1661488338/analytics_re_wxt8_1920e22fe3.svg',
                  previewUrl: null,
                  provider: 'cloudinary',
                  provider_metadata: {
                    public_id: 'analytics_re_wxt8_1920e22fe3',
                    resource_type: 'image',
                  },
                  createdAt: '2022-08-26T04:34:28.247Z',
                  updatedAt: '2022-08-26T04:34:28.247Z',
                },
              },
            },
            metadata: {
              id: 1,
              name: 'index',
              section_id: 'index',
              background: true,
            },
          },
          {
            id: 1,
            __component: 'section.section-content',
            title: 'Titulo qualquer',
            content:
              "Lorem Ipsum is **simply dummy t**ext of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since _the 1500s, _when an unknown printer\n\nLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer\n\nLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer",
            metadata: {
              id: 3,
              name: 'why-us',
              section_id: 'why-us',
              background: false,
            },
          },
          {
            id: 1,
            __component: 'section.section-grid',
            title: 'works',
            description: 'all works',
            metadata: {
              id: 2,
              name: 'grid-one',
              section_id: 'grid-one',
              background: true,
            },
            text_grid: [
              {
                title: 'titulo 1',
                description:
                  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
              },
              {
                title: 'titulo 2',
                description:
                  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
              },
              {
                title: 'titulo 3',
                description:
                  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
              },
            ],
            image_grid: {},
          },
        ],
        menu: {
          id: 1,
          logo_link: '#index',
          logo_text: 'Logo',
          image: {
            data: {
              id: 3,
              attributes: {
                name: 'landing_page_2_c6baa21725.svg',
                alternativeText: 'landing_page_2_c6baa21725.svg',
                caption: 'landing_page_2_c6baa21725.svg',
                width: 47,
                height: 9,
                formats: null,
                hash: 'landing_page_2_c6baa21725_db7d79cda1',
                ext: '.svg',
                mime: 'image/svg+xml',
                size: 2.51,
                url: 'https://res.cloudinary.com/brianronin/image/upload/v1661544963/landing_page_2_c6baa21725_db7d79cda1.svg',
                previewUrl: null,
                provider: 'cloudinary',
                provider_metadata: {
                  public_id: 'landing_page_2_c6baa21725_db7d79cda1',
                  resource_type: 'image',
                },
                createdAt: '2022-08-26T20:18:16.375Z',
                updatedAt: '2022-08-26T20:18:16.375Z',
              },
            },
          },
          menu_links: [
            {
              id: 2,
              link_text: 'why us',
              urll: '#why-us',
              open_in_new_tab: false,
            },
            {
              id: 3,
              link_text: 'grid one',
              urll: '#grid-one',
              open_in_new_tab: false,
            },
            {
              id: 1,
              link_text: 'intro',
              urll: '#index',
              open_in_new_tab: false,
            },
          ],
        },
      },
    },
    {
      id: 3,
      attributes: {
        title: 'minha pagina2',
        slug: 'minha-pagina2',
        createdAt: '2022-08-26T20:52:34.873Z',
        updatedAt: '2022-09-10T01:16:46.805Z',
        publishedAt: '2022-08-26T20:52:54.889Z',
        footer_text: 'feito com carinho por BrianRonin',
        sections: [
          {
            id: 2,
            __component: 'section.section-two-columns',
            title: 'titulo qualquer',
            description: 'uma breve desc',
            image: {
              data: {
                id: 2,
                attributes: {
                  name: 'analytics_re_wxt8.svg',
                  alternativeText: 'analytics_re_wxt8.svg',
                  caption: 'analytics_re_wxt8.svg',
                  width: 363,
                  height: 438,
                  formats: null,
                  hash: 'analytics_re_wxt8_1920e22fe3',
                  ext: '.svg',
                  mime: 'image/svg+xml',
                  size: 7.3,
                  url: 'https://res.cloudinary.com/brianronin/image/upload/v1661488338/analytics_re_wxt8_1920e22fe3.svg',
                  previewUrl: null,
                  provider: 'cloudinary',
                  provider_metadata: {
                    public_id: 'analytics_re_wxt8_1920e22fe3',
                    resource_type: 'image',
                  },
                  createdAt: '2022-08-26T04:34:28.247Z',
                  updatedAt: '2022-08-26T04:34:28.247Z',
                },
              },
            },
            metadata: {
              id: 5,
              name: 'index',
              section_id: 'index',
              background: true,
            },
          },
          {
            id: 2,
            __component: 'section.section-content',
            title: 'Titulo qualquer',
            content:
              "Lorem Ipsum is **simply dummy t**ext of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since _the 1500s, _when an unknown printer\n\nLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer\n\nLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer",
            metadata: {
              id: 4,
              name: 'why-us',
              section_id: 'why-us',
              background: false,
            },
          },
          {
            id: 2,
            __component: 'section.section-grid',
            title: 'works',
            description: 'all works',
            metadata: {
              id: 6,
              name: 'grid-one',
              section_id: 'grid-one',
              background: true,
            },
            text_grid: [],
            image_grid: {
              id: 1,
              image: {
                data: [
                  {
                    id: 5,
                    attributes: {
                      name: '800x800',
                      alternativeText: '800x800',
                      caption: '800x800',
                      width: 800,
                      height: 800,
                      formats: {
                        small: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772483/small_800x800_78230a377d.jpg',
                          hash: 'small_800x800_78230a377d',
                          mime: 'image/jpeg',
                          name: 'small_800x800',
                          path: null,
                          size: 58.11,
                          width: 500,
                          height: 500,
                          provider_metadata: {
                            public_id: 'small_800x800_78230a377d',
                            resource_type: 'image',
                          },
                        },
                        medium: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772483/medium_800x800_78230a377d.jpg',
                          hash: 'medium_800x800_78230a377d',
                          mime: 'image/jpeg',
                          name: 'medium_800x800',
                          path: null,
                          size: 127.55,
                          width: 750,
                          height: 750,
                          provider_metadata: {
                            public_id: 'medium_800x800_78230a377d',
                            resource_type: 'image',
                          },
                        },
                        thumbnail: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772482/thumbnail_800x800_78230a377d.jpg',
                          hash: 'thumbnail_800x800_78230a377d',
                          mime: 'image/jpeg',
                          name: 'thumbnail_800x800',
                          path: null,
                          size: 6.24,
                          width: 156,
                          height: 156,
                          provider_metadata: {
                            public_id: 'thumbnail_800x800_78230a377d',
                            resource_type: 'image',
                          },
                        },
                      },
                      hash: '800x800_78230a377d',
                      ext: '.jpeg',
                      mime: 'image/jpeg',
                      size: 145.57,
                      url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772483/800x800_78230a377d.jpg',
                      previewUrl: null,
                      provider: 'cloudinary',
                      provider_metadata: {
                        public_id: '800x800_78230a377d',
                        resource_type: 'image',
                      },
                      createdAt: '2022-09-10T01:14:43.471Z',
                      updatedAt: '2022-09-10T01:14:43.471Z',
                    },
                  },
                  {
                    id: 6,
                    attributes: {
                      name: '800x800',
                      alternativeText: '800x800',
                      caption: '800x800',
                      width: 800,
                      height: 800,
                      formats: {
                        small: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772500/small_800x800_6c695611a0.jpg',
                          hash: 'small_800x800_6c695611a0',
                          mime: 'image/jpeg',
                          name: 'small_800x800',
                          path: null,
                          size: 56.69,
                          width: 500,
                          height: 500,
                          provider_metadata: {
                            public_id: 'small_800x800_6c695611a0',
                            resource_type: 'image',
                          },
                        },
                        medium: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772500/medium_800x800_6c695611a0.jpg',
                          hash: 'medium_800x800_6c695611a0',
                          mime: 'image/jpeg',
                          name: 'medium_800x800',
                          path: null,
                          size: 114.41,
                          width: 750,
                          height: 750,
                          provider_metadata: {
                            public_id: 'medium_800x800_6c695611a0',
                            resource_type: 'image',
                          },
                        },
                        thumbnail: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772500/thumbnail_800x800_6c695611a0.jpg',
                          hash: 'thumbnail_800x800_6c695611a0',
                          mime: 'image/jpeg',
                          name: 'thumbnail_800x800',
                          path: null,
                          size: 7.45,
                          width: 156,
                          height: 156,
                          provider_metadata: {
                            public_id: 'thumbnail_800x800_6c695611a0',
                            resource_type: 'image',
                          },
                        },
                      },
                      hash: '800x800_6c695611a0',
                      ext: '.jpeg',
                      mime: 'image/jpeg',
                      size: 127.95,
                      url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772500/800x800_6c695611a0.jpg',
                      previewUrl: null,
                      provider: 'cloudinary',
                      provider_metadata: {
                        public_id: '800x800_6c695611a0',
                        resource_type: 'image',
                      },
                      createdAt: '2022-09-10T01:15:01.274Z',
                      updatedAt: '2022-09-10T01:15:01.274Z',
                    },
                  },
                  {
                    id: 7,
                    attributes: {
                      name: '800x800',
                      alternativeText: '800x800',
                      caption: '800x800',
                      width: 800,
                      height: 800,
                      formats: {
                        small: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772516/small_800x800_9267725d72.jpg',
                          hash: 'small_800x800_9267725d72',
                          mime: 'image/jpeg',
                          name: 'small_800x800',
                          path: null,
                          size: 39.52,
                          width: 500,
                          height: 500,
                          provider_metadata: {
                            public_id: 'small_800x800_9267725d72',
                            resource_type: 'image',
                          },
                        },
                        medium: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772516/medium_800x800_9267725d72.jpg',
                          hash: 'medium_800x800_9267725d72',
                          mime: 'image/jpeg',
                          name: 'medium_800x800',
                          path: null,
                          size: 80.99,
                          width: 750,
                          height: 750,
                          provider_metadata: {
                            public_id: 'medium_800x800_9267725d72',
                            resource_type: 'image',
                          },
                        },
                        thumbnail: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772516/thumbnail_800x800_9267725d72.jpg',
                          hash: 'thumbnail_800x800_9267725d72',
                          mime: 'image/jpeg',
                          name: 'thumbnail_800x800',
                          path: null,
                          size: 4.65,
                          width: 156,
                          height: 156,
                          provider_metadata: {
                            public_id: 'thumbnail_800x800_9267725d72',
                            resource_type: 'image',
                          },
                        },
                      },
                      hash: '800x800_9267725d72',
                      ext: '.jpeg',
                      mime: 'image/jpeg',
                      size: 90.49,
                      url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772516/800x800_9267725d72.jpg',
                      previewUrl: null,
                      provider: 'cloudinary',
                      provider_metadata: {
                        public_id: '800x800_9267725d72',
                        resource_type: 'image',
                      },
                      createdAt: '2022-09-10T01:15:16.769Z',
                      updatedAt: '2022-09-10T01:15:16.769Z',
                    },
                  },
                  {
                    id: 4,
                    attributes: {
                      name: '800x800',
                      alternativeText: '800x800',
                      caption: '800x800',
                      width: 800,
                      height: 800,
                      formats: {
                        small: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772355/small_800x800_ee870b345a.jpg',
                          hash: 'small_800x800_ee870b345a',
                          mime: 'image/jpeg',
                          name: 'small_800x800',
                          path: null,
                          size: 18.19,
                          width: 500,
                          height: 500,
                          provider_metadata: {
                            public_id: 'small_800x800_ee870b345a',
                            resource_type: 'image',
                          },
                        },
                        medium: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772355/medium_800x800_ee870b345a.jpg',
                          hash: 'medium_800x800_ee870b345a',
                          mime: 'image/jpeg',
                          name: 'medium_800x800',
                          path: null,
                          size: 35.86,
                          width: 750,
                          height: 750,
                          provider_metadata: {
                            public_id: 'medium_800x800_ee870b345a',
                            resource_type: 'image',
                          },
                        },
                        thumbnail: {
                          ext: '.jpeg',
                          url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772354/thumbnail_800x800_ee870b345a.jpg',
                          hash: 'thumbnail_800x800_ee870b345a',
                          mime: 'image/jpeg',
                          name: 'thumbnail_800x800',
                          path: null,
                          size: 2.2,
                          width: 156,
                          height: 156,
                          provider_metadata: {
                            public_id: 'thumbnail_800x800_ee870b345a',
                            resource_type: 'image',
                          },
                        },
                      },
                      hash: '800x800_ee870b345a',
                      ext: '.jpeg',
                      mime: 'image/jpeg',
                      size: 39.22,
                      url: 'https://res.cloudinary.com/brianronin/image/upload/v1662772355/800x800_ee870b345a.jpg',
                      previewUrl: null,
                      provider: 'cloudinary',
                      provider_metadata: {
                        public_id: '800x800_ee870b345a',
                        resource_type: 'image',
                      },
                      createdAt: '2022-09-10T01:12:35.264Z',
                      updatedAt: '2022-09-10T01:12:35.264Z',
                    },
                  },
                ],
              },
            },
          },
        ],
        menu: {
          id: 2,
          logo_link: '#index',
          logo_text: 'Logo',
          image: {
            data: {
              id: 3,
              attributes: {
                name: 'landing_page_2_c6baa21725.svg',
                alternativeText: 'landing_page_2_c6baa21725.svg',
                caption: 'landing_page_2_c6baa21725.svg',
                width: 47,
                height: 9,
                formats: null,
                hash: 'landing_page_2_c6baa21725_db7d79cda1',
                ext: '.svg',
                mime: 'image/svg+xml',
                size: 2.51,
                url: 'https://res.cloudinary.com/brianronin/image/upload/v1661544963/landing_page_2_c6baa21725_db7d79cda1.svg',
                previewUrl: null,
                provider: 'cloudinary',
                provider_metadata: {
                  public_id: 'landing_page_2_c6baa21725_db7d79cda1',
                  resource_type: 'image',
                },
                createdAt: '2022-08-26T20:18:16.375Z',
                updatedAt: '2022-08-26T20:18:16.375Z',
              },
            },
          },
          menu_links: [
            {
              id: 4,
              link_text: 'why us',
              urll: '#why-us',
              open_in_new_tab: false,
            },
            {
              id: 6,
              link_text: 'grid one',
              urll: '#grid-one',
              open_in_new_tab: false,
            },
            {
              id: 5,
              link_text: 'intro',
              urll: '#index',
              open_in_new_tab: false,
            },
          ],
        },
      },
    },
  ],
  meta: {
    pagination: {
      page: 1,
      pageSize: 25,
      pageCount: 1,
      total: 2,
    },
  },
}
