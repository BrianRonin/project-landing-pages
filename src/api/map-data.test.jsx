import { mapData } from './map-data'
import { mockApi } from './mockApi2'
const util = require('util')
describe('map-data', () => {
  it('should return correct values', () => {
    const map = mapData(mockApi)
    expect(map).toMatchSnapshot()
    const mapNo = mapData()
    expect(mapNo).toEqual([
      {
        footerHtml: '',
        slug: '',
        title: '',
        sections: [],
        menu: { link: '', text: '', srcImage: {}, links: [] },
      },
    ])

    // const util = require('util')
    // console.log(util.inspect(mapNo, false, null, true /* enable colors */))

    const data = [
      ({
        footerHtml: 'feito com carinho por BrianRonin',
        slug: 'minha-pagina',
        title: 'minha pagina',
        sections: [
          {
            component: 'section.section-two-columns',
            title: 'titulo qualquer',
            text: 'uma breve desc',
            srcImg:
              'https://res.cloudinary.com/brianronin/image/upload/v1661488338/analytics_re_wxt8_1920e22fe3.svg',
            name: 'index',
            id: 'index',
            background: true,
          },
          {
            title: 'Titulo qualquer',
            html:
              "Lorem Ipsum is **simply dummy t**ext of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since _the 1500s, _when an unknown printer\n" +
              '\n' +
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer\n" +
              '\n' +
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer",
            name: 'why-us',
            id: 'why-us',
            background: false,
          },
          {
            title: 'works',
            description: 'all works',
            grid: [
              {
                title: 'titulo 1',
                description:
                  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
              },
              {
                title: 'titulo 2',
                description:
                  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
              },
              {
                title: 'titulo 3',
                description:
                  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
              },
            ],
            name: 'grid-one',
            id: 'grid-one',
            background: true,
          },
        ],
        menu: {
          link: '#index',
          text: 'Logo',
          srcImage:
            'https://res.cloudinary.com/brianronin/image/upload/v1661544963/landing_page_2_c6baa21725_db7d79cda1.svg',
          links: [
            { newTab: false, children: 'why us', Link: '#why-us' },
            { newTab: false, children: 'grid one', Link: '#grid-one' },
            { newTab: false, children: 'intro', Link: '#index' },
          ],
        },
      },
      {
        footerHtml: 'feito com carinho por BrianRonin',
        slug: 'minha-pagina2',
        title: 'minha pagina2',
        sections: [
          {
            component: 'section.section-two-columns',
            title: 'titulo qualquer',
            text: 'uma breve desc',
            srcImg:
              'https://res.cloudinary.com/brianronin/image/upload/v1661488338/analytics_re_wxt8_1920e22fe3.svg',
            name: 'index',
            id: 'index',
            background: true,
          },
          {
            title: 'Titulo qualquer',
            html:
              "Lorem Ipsum is **simply dummy t**ext of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since _the 1500s, _when an unknown printer\n" +
              '\n' +
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer\n" +
              '\n' +
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer",
            name: 'why-us',
            id: 'why-us',
            background: false,
          },
          {
            title: 'works',
            description: 'all works',
            grid: [
              'https://res.cloudinary.com/brianronin/image/upload/v1662772483/800x800_78230a377d.jpg',
              'https://res.cloudinary.com/brianronin/image/upload/v1662772500/800x800_6c695611a0.jpg',
              'https://res.cloudinary.com/brianronin/image/upload/v1662772516/800x800_9267725d72.jpg',
              'https://res.cloudinary.com/brianronin/image/upload/v1662772355/800x800_ee870b345a.jpg',
            ],
            name: 'grid-one',
            id: 'grid-one',
            background: true,
          },
        ],
        menu: {
          link: '#index',
          text: 'Logo',
          srcImage:
            'https://res.cloudinary.com/brianronin/image/upload/v1661544963/landing_page_2_c6baa21725_db7d79cda1.svg',
          links: [
            { newTab: false, children: 'why us', Link: '#why-us' },
            { newTab: false, children: 'grid one', Link: '#grid-one' },
            { newTab: false, children: 'intro', Link: '#index' },
          ],
        },
      }),
    ]
  })
})
