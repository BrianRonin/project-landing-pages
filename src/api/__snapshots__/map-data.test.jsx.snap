// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map-data should return correct values 1`] = `
Array [
  Object {
    "footerHtml": "feito com carinho por BrianRonin",
    "menu": Object {
      "link": "#index",
      "links": Array [
        Object {
          "Link": "#why-us",
          "children": "why us",
          "newTab": false,
        },
        Object {
          "Link": "#grid-one",
          "children": "grid one",
          "newTab": false,
        },
        Object {
          "Link": "#index",
          "children": "intro",
          "newTab": false,
        },
      ],
      "srcImage": "https://res.cloudinary.com/brianronin/image/upload/v1661544963/landing_page_2_c6baa21725_db7d79cda1.svg",
      "text": "Logo",
    },
    "sections": Array [
      Object {
        "background": true,
        "component": "section.section-two-columns",
        "id": "index",
        "name": "index",
        "srcImg": "https://res.cloudinary.com/brianronin/image/upload/v1661488338/analytics_re_wxt8_1920e22fe3.svg",
        "text": "uma breve desc",
        "title": "titulo qualquer",
      },
      Object {
        "background": false,
        "html": "Lorem Ipsum is **simply dummy t**ext of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since _the 1500s, _when an unknown printer

Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer

Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer",
        "id": "why-us",
        "name": "why-us",
        "title": "Titulo qualquer",
      },
      Object {
        "background": true,
        "description": "all works",
        "grid": Array [
          Object {
            "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
            "title": "titulo 1",
          },
          Object {
            "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
            "title": "titulo 2",
          },
          Object {
            "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make ",
            "title": "titulo 3",
          },
        ],
        "id": "grid-one",
        "name": "grid-one",
        "title": "works",
      },
    ],
    "slug": "minha-pagina",
    "title": "minha pagina",
  },
  Object {
    "footerHtml": "feito com carinho por BrianRonin",
    "menu": Object {
      "link": "#index",
      "links": Array [
        Object {
          "Link": "#why-us",
          "children": "why us",
          "newTab": false,
        },
        Object {
          "Link": "#grid-one",
          "children": "grid one",
          "newTab": false,
        },
        Object {
          "Link": "#index",
          "children": "intro",
          "newTab": false,
        },
      ],
      "srcImage": "https://res.cloudinary.com/brianronin/image/upload/v1661544963/landing_page_2_c6baa21725_db7d79cda1.svg",
      "text": "Logo",
    },
    "sections": Array [
      Object {
        "background": true,
        "component": "section.section-two-columns",
        "id": "index",
        "name": "index",
        "srcImg": "https://res.cloudinary.com/brianronin/image/upload/v1661488338/analytics_re_wxt8_1920e22fe3.svg",
        "text": "uma breve desc",
        "title": "titulo qualquer",
      },
      Object {
        "background": false,
        "html": "Lorem Ipsum is **simply dummy t**ext of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since _the 1500s, _when an unknown printer

Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer

Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer",
        "id": "why-us",
        "name": "why-us",
        "title": "Titulo qualquer",
      },
      Object {
        "background": true,
        "description": "all works",
        "grid": Array [
          "https://res.cloudinary.com/brianronin/image/upload/v1662772483/800x800_78230a377d.jpg",
          "https://res.cloudinary.com/brianronin/image/upload/v1662772500/800x800_6c695611a0.jpg",
          "https://res.cloudinary.com/brianronin/image/upload/v1662772516/800x800_9267725d72.jpg",
          "https://res.cloudinary.com/brianronin/image/upload/v1662772355/800x800_ee870b345a.jpg",
        ],
        "id": "grid-one",
        "name": "grid-one",
        "title": "works",
      },
    ],
    "slug": "minha-pagina2",
    "title": "minha pagina2",
  },
]
`;
